metricq/cli/wrapper.py

```diff
@@ -0,0 +1,106 @@
+import logging
+from typing import Callable, Optional, cast
+
+import click
+import click_log  # type: ignore
+from click import option
+from dotenv import find_dotenv, load_dotenv
+
+from .. import get_logger
+from .params import MetricParam, TemplateStringParam
+from .types import FC
+
+# We do not interpolate (i.e. replace ${VAR} with corresponding environment variables).
+# That is because we want to be able to interpolate ourselves for metrics and tokens
+# using the same syntax. If it was only ${USER} for the token, we could use the
+# override functionality, but most unfortunately there is no standard environment
+# variable for the hostname. Even $HOST on zsh is not actually part of the environment.
+# ``override=false`` just means that environment variables have priority over the
+# env files.
+load_dotenv(dotenv_path=find_dotenv(".metricq"), interpolate=False, override=False)
+
+
+def metricq_server_option() -> Callable[[FC], FC]:
+    return option(
+        "--server",
+        type=TemplateStringParam(),
+        metavar="URL",
+        required=True,
+        help="MetricQ server URL.",
+    )
+
+
+def metricq_token_option(default: str) -> Callable[[FC], FC]:
+    return option(
+        "--token",
+        type=TemplateStringParam(),
+        metavar="CLIENT_TOKEN",
+        default=default,
+        show_default=True,
+        help="A token to identify this client on the MetricQ network.",
+    )
+
+
+def metricq_metric_option(
+    default: Optional[str] = None, multiple: bool = False
+) -> Callable[[FC], FC]:
+    response_default = default if (default is None or not multiple) else [default]
+    help = "Use the -–metric / -m parameter to specify which metric the program should use."
+    if default:
+        help += "You can also specify multpile metrics."
+
+    return option(
+        "--metric",
+        "-m",
+        type=MetricParam(),
+        metavar="METRIC",
+        show_default=True,
+        required=default is None,
```

Comment:
Thinking about it, a parameter can have no default and be not required. That's called optional. We should support that. Please add a `required` parameter to the decorator.